titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/internal/JobFeatureComplianceChecks.java

```diff
@@ -0,0 +1,140 @@
+                return Optional.empty();
+            }
+            return Optional.of(NonComplianceSet.of(
+                    IAM_ROLE_REQUIRED_FEATURE,
+                    jobDescriptor,
+                    NO_IAM_ROLE_CONTEXT,
+                    "IAM role not set in the job descriptor"
+            ));
+        };
+    }
+
+    /**
+     * See {@link FeatureRolloutPlans#SECURITY_GROUPS_REQUIRED_FEATURE}.
+     */
+    static FeatureCompliance<JobDescriptor<?>> missingSecurityGroups() {
+        return jobDescriptor -> {
+            if (!jobDescriptor.getContainer().getSecurityProfile().getSecurityGroups().isEmpty()) {
+                return Optional.empty();
+            }
+            return Optional.of(NonComplianceSet.of(
+                    SECURITY_GROUPS_REQUIRED_FEATURE,
+                    jobDescriptor,
+                    NO_SECURITY_GROUPS_CONTEXT,
+                    "At least one security group must be set in the job descriptor"
+            ));
+        };
+    }
+
+    /**
+     * See {@link FeatureRolloutPlans#ENTRY_POINT_STRICT_VALIDATION_FEATURE}.
+     */
+    static FeatureCompliance<JobDescriptor<?>> entryPointViolations() {
+        return jobDescriptor -> {
+            List<String> entryPoint = jobDescriptor.getContainer().getEntryPoint();
+            List<String> command = jobDescriptor.getContainer().getCommand();
+            if (!CollectionsExt.isNullOrEmpty(entryPoint) && CollectionsExt.isNullOrEmpty(command) && CONTAINS_SPACES.test(entryPoint.get(0))) {
+                return Optional.of(NonComplianceSet.of(
+                        ENTRY_POINT_STRICT_VALIDATION_FEATURE,
+                        jobDescriptor,
+                        ENTRY_POINT_WITH_SPACES_CONTEXT,
+                        "First entry point value cannot contain spaces"
+                ));
+            }
+            return Optional.empty();
+        };
+    }
+
+    /**
+     * See {@link FeatureRolloutPlans#MIN_DISK_SIZE_STRICT_VALIDATION_FEATURE}.
+     */
+    static FeatureCompliance<JobDescriptor<?>> minDiskSize(JobManagerConfiguration jobManagerConfiguration) {
+        return jobDescriptor -> {
+
+            ContainerResources containerResources = jobDescriptor.getContainer().getContainerResources();
+            int minDiskSize = jobManagerConfiguration.getMinDiskSizeMB();
+            if (containerResources.getDiskMB() >= minDiskSize) {
+                return Optional.empty();
+            }
+
+            return Optional.of(NonComplianceSet.of(
+                    MIN_DISK_SIZE_STRICT_VALIDATION_FEATURE,
+                    jobDescriptor,
+                    Collections.singletonMap("diskSizeLessThanMin", String.format("Minimum disk size is %sMB, but is set %sMB", minDiskSize, containerResources.getDiskMB())),
+                    String.format("Job descriptor must declare disk size that is no less than %sMB", containerResources.getDiskMB())
```

Comment:
shouldn't `containerResources.getDiskMB()` be `minDiskSize`?

Reply:
Yes. Fixed.